### PR TITLE
Updated lookup tables structure

### PR DESCRIPTION
**Pull request by Katlegow**
Hey guys - kindly review for me there.

Thanks.

### CODE REVIEW

File: migrations/mirror/2021-09-22-add-acive-key-to-lookup-tables.sql [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+ALTER TABLE "refProject_ProjectState" ADD COLUMN "deactivated_at" TIMESTAMP WITHOUT TIME ZONE;
+ALTER TABLE "refProject_ProjectState" ADD COLUMN "allow_delete" BOOLEAN;
+CREATE UNIQUE INDEX ON "refProject_ProjectState" ("guid");
+CREATE INDEX ON "refProject_ProjectState" ("v");
+
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+
+ALTER TABLE public."refActual_ExpenditureType" ADD CONSTRAINT check_chars CHECK (k::text <> ''::text AND k::text ~ '^[\w]*$'::text);
+ALTER TABLE public."refActual_ExpenditureType" ALTER COLUMN guid SET DEFAULT uuid_generate_v4();
+ALTER TABLE public."refActual_ExpenditureType" ALTER COLUMN creation_date SET DEFAULT now();
```
Comment (FritzOnFire) [inline migrations/mirror/2021-09-22-add-acive-key-to-lookup-tables.sql:138]: I don't think these defaults work the way you think they work https://www.postgresql.org/docs/10/ddl-alter.html#id-1.5.4.7.9 . I think you rather want to add the default to the `ADD COLUMN` statement.

**Review by FritzOnFire — CHANGES_REQUESTED** [review]
I think you just need to change the way you add the defaults, and then this should be good.

**Review by FritzOnFire — APPROVED** [review]
LGTM

File: migrations/mirror/2021-09-22-add-acive-key-to-lookup-tables.sql [hunk 2]
```diff
@@ -0,0 +1,222 @@
+ALTER TABLE "refActual_ExpenditureType" ADD COLUMN "guid" UUID;
+ALTER TABLE "refActual_ExpenditureType" ADD COLUMN "creation_date" TIMESTAMP WITHOUT TIME ZONE;
+ALTER TABLE "refActual_ExpenditureType" ADD COLUMN "deactivated_at" TIMESTAMP WITHOUT TIME ZONE;
+ALTER TABLE "refActual_ExpenditureType" ADD COLUMN "allow_delete" BOOLEAN;
+CREATE UNIQUE INDEX ON "refActual_ExpenditureType" ("guid");
```
Comment (frankvanrietimqs) [inline migrations/mirror/2021-09-22-add-acive-key-to-lookup-tables.sql:7]: Wowee. I would add some blank lines between tables

File: migrations/mirror/2021-09-22-add-acive-key-to-lookup-tables.sql [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+CREATE INDEX ON "refActual_ExpenditureType" ("v");
+ALTER TABLE "refBOQ_Unit" ADD COLUMN "guid" UUID;
+ALTER TABLE "refBOQ_Unit" ADD COLUMN "creation_date" TIMESTAMP WITHOUT TIME ZONE;
+ALTER TABLE "refBOQ_Unit" ADD COLUMN "deactivated_at" TIMESTAMP WITHOUT TIME ZONE;
+ALTER TABLE "refBOQ_Unit" ADD COLUMN "allow_delete" BOOLEAN;
+CREATE UNIQUE INDEX ON "refBOQ_Unit" ("guid");
```
Comment (frankvanrietimqs) [inline migrations/mirror/2021-09-22-add-acive-key-to-lookup-tables.sql:13]: For future reference there is no need to quote all lower case identifiers unless they are reserved words.

File: migrations/mirror/2021-09-22-add-acive-key-to-lookup-tables.sql [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+CREATE INDEX ON "refBOQ_Unit" ("v");
+ALTER TABLE "refBudget_BudgetType" ADD COLUMN "guid" UUID;
+ALTER TABLE "refBudget_BudgetType" ADD COLUMN "creation_date" TIMESTAMP WITHOUT TIME ZONE;
+ALTER TABLE "refBudget_BudgetType" ADD COLUMN "deactivated_at" TIMESTAMP WITHOUT TIME ZONE;
+ALTER TABLE "refBudget_BudgetType" ADD COLUMN "allow_delete" BOOLEAN;
```
Comment (frankvanrietimqs) [inline migrations/mirror/2021-09-22-add-acive-key-to-lookup-tables.sql:18]: Check the spec, but I am pretty sure allow_delete should default to false

File: migrations/mirror/2021-09-22-add-acive-key-to-lookup-tables.sql [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+CREATE UNIQUE INDEX ON "refProject_FinSysProjectState" ("guid");
+CREATE INDEX ON "refProject_FinSysProjectState" ("v");
+ALTER TABLE "refProject_ProjectState" ADD COLUMN "guid" UUID DEFAULT uuid_generate_v4();
+ALTER TABLE "refProject_ProjectState" ADD COLUMN "creation_date" TIMESTAMP WITHOUT TIME ZONE DEFAULT now();
+ALTER TABLE "refProject_ProjectState" ADD COLUMN "deactivated_at" TIMESTAMP WITHOUT TIME ZONE;
+ALTER TABLE "refProject_ProjectState" ADD COLUMN "allow_delete" BOOLEAN;
+CREATE UNIQUE INDEX ON "refProject_ProjectState" ("guid");
+CREATE INDEX ON "refProject_ProjectState" ("v");
+
+ALTER TABLE public."refActual_ExpenditureType" ADD CONSTRAINT check_chars CHECK (k::text <> ''::text AND k::text ~ '^[\w]*$'::text);
```
Comment (frankvanrietimqs) [inline migrations/mirror/2021-09-22-add-acive-key-to-lookup-tables.sql:136]: No need to change it now, but I would suggest that in future you group all changes to a given table. With such a large file it is very tedious to have to scroll up and down the whole time

File: migrations/mirror/2021-09-22-add-acive-key-to-lookup-tables.sql (hunk 4 above)
Comment (Katlegow) [reply migrations/mirror/2021-09-22-add-acive-key-to-lookup-tables.sql:18]: According to the spec null implied false but it make sense to default it false. Will fix it.

File: migrations/mirror/2021-09-22-add-acive-key-to-lookup-tables.sql (hunk 5 above)
Comment (Katlegow) [reply migrations/mirror/2021-09-22-add-acive-key-to-lookup-tables.sql:136]: I have all the time at the moment so I will just change it now.

File: migrations/mirror/2021-09-22-add-acive-key-to-lookup-tables.sql (hunk 2 above)
Comment (FritzOnFire) [reply migrations/mirror/2021-09-22-add-acive-key-to-lookup-tables.sql:7]: Your commenting on auto generated code...

**Pull request closed without merging**